Upgrade dependencies, switch to ESLint

alehuo opened the pull request:
- Update .nvmrc to use Node v12.17.0
- Switch to npm
- Remove suffixes from db names from all except test env
- Delete yarn.lock
- Switch TSLint to ESLint
- Upgrade TypeScript to 4.0.3.
- Add typecheck command
- Add ECR deploy stage to deploy.yml

cxcorp commented on Dockerfile:
```diff
@@ -1,18 +1,16 @@
-COPY yarn.lock package.json /app/
-RUN yarn --dev --frozen-lockfile
+COPY package*.json /app/
+RUN npm install --development
```
btw what are the dev deps used for?

alehuo replied:
Dev dependencies are used in the build phase, now they are installed with npm instead of yarn

cxcorp commented on Dockerfile:
```diff
@@ -1,18 +1,16 @@
 EXPOSE 3001
 
-CMD ["yarn", "start"]
+CMD ["npm", "start"]
```
any reason why we use `ts-node` instead of building it and starting the dist directly with `node`?

alehuo replied:
I have the same question as you. `ts-node` is just `node` + ´tsc´. I think it's better to use the native node binary instead of `ts-node` to run user service. `ts-node` has increased memory usage compared to `node`, and it's also slower than `node`. Also, the bridge between build- and run-time is not that clear with `ts-node`. Better to first build the project with `tsc`, which will emit errors straight away.

cxcorp replied:
yeah + fewer deps to include in the output image

cxcorp replied:
@MergHQ any reason for ts-node over tsc at image build time?

MergHQ replied:
Emmä vittu tiedä, kai laiskuus on se syy

cxcorp replied:
iha ok

MergHQ replied:
Mut hyvä idea olla käyttämättä ts-nodee prodissa

cxcorp commented:
What was the "Remove suffixes from db names from all except test env" change and what was it for? What were the db suffixes for anyway?

cxcorp commented:
In general yea lgtm :+1: merge it when ready. I'd prefer individual commits, especially for those that rerun prettier cos then it's easier to see what _actually_ changed instead of was just reformatted.

alehuo commented:
The quotes in DB names were used to differentiate the databases used for development, test and production environments. This was a "safeguard" for a situation where tests were run against production database or the production database was used for the wrong environment.

This is a good practice, but user service runs in a container which is passed the correct environment vars and only starts the app.

alehuo commented:
More changes:
- SASS files, localizations and public folder is now resovled by `process.cwd()` which is the same used by Express in resolving the views folder
- Updated Dockerfile
- Modified `start` script in `package.json`

MergHQ reviewed (approved):
Fix conflicts, then LGTM
